File: ruby/004.rb
# Find the largest palindrome made from the product of two 3-digit numbers.

def palindrome? (n)
    n.to_s == n.to_s.reverse
end

largest_palind = 0
(100..999).each do |x|
    (100..999).each do |y|
        mult = x*y
        largest_palind = mult if palindrome?(mult) and mult > largest_palind
    end
end
puts largest_palind

File: clojure/src/project_euler_clojure/004.clj
;; Find the largest palindrome made from the product of two 3-digit numbers.

(ns project-euler-clojure.004)

(defn products-3-digit-numbers []
  (for [x (range 100 1000)
        y (range 100 1000)]
    (* x y)))

(defn palindrome? [n]
  (= (reverse (str n))
     (seq (str n))))

(defn problem-004 []
  (->> (products-3-digit-numbers)
       (filter palindrome?)
       (apply max)))
